lib/ui/window/platform_configuration.cc

```diff
@@ -77,12 +76,34 @@ void PlatformConfiguration::DidCreateIsolate() {
   report_timings_.Set(tonic::DartState::Current(),
                       Dart_GetField(library, tonic::ToDart("_reportTimings")));
 
-  // TODO(loicsharma): This should only be created if the embedder enables the
-  // implicit view.
-  // See: https://github.com/flutter/flutter/issues/120306
-  windows_.emplace(kImplicitViewId,
-                   std::make_unique<Window>(
-                       kImplicitViewId, ViewportMetrics{1.0, 0.0, 0.0, -1, 0}));
+  library_.Set(tonic::DartState::Current(), library);
+}
+
+void PlatformConfiguration::AddView(int64_t view_id,
+                                    const ViewportMetrics& view_metrics) {
+  auto [window_iterator, insertion_happened] = windows_.emplace(
+      view_id, std::make_unique<Window>(library_, view_id, view_metrics));
+  FML_DCHECK(insertion_happened);
+  // Make the new window send an AddView message to Dart.
+  window_iterator->second->AddView();
+}
+
+void PlatformConfiguration::RemoveView(int64_t view_id) {
+  FML_DCHECK(view_id != kFlutterImplicitViewId)
+      << "The implicit view #" << view_id << " should never be removed.";
+  size_t erased_elements = windows_.erase(view_id);
+  FML_DCHECK(erased_elements != 0) << "View #" << view_id << " doesn't exist.";
```

Comment:
Should these be DCHECK or CHECKs?

DCHECK will only run in a local unopt engine build. Flutter developers will never see these.

It seems like this should always be an error log at the least. Alternatively, it could be a DCHECK and an error log with a fizzle, e.g.

```cpp
if (view_id == kFlutterImplicitViewId) {
  FML_LOG(ERROR) << "The implicit view #" << view_id << " cannot be removed.";
  FML_DCHECK(false);
  return;
}
```

Reply:
Similar comment for the other DCHECK on 95

Reply:
I've changed one of them to the "if" pattern you suggested. I think both should stay DCHECK since they should never happen if `Shell` is implemented correctly, and serve as debug-only assertions. I changed the first check to an `if` to prevent the implicit view being deleted, but didn't for the second check so to make sure the message is always sent to the Dart VM